experimental/dds/tree2/src/feature-libraries/sequence-field/rebase.ts

```diff
@@ -423,258 +327,145 @@ interface RebaseMarks<T> {
-					if (newCurrMark.isIntention) {
-						newCurrMark.detachedBy = baseIntention;
-					} else if (baseIntention !== newCurrMark.detachedBy) {
-						newCurrMark.lastDetachedBy = baseIntention;
-					}
-					newCurrMark.detachIndex = baseInputOffset;
-					delete (newCurrMark as CanConflict).conflictsWith;
-					const effect = getOrAddEffect(
-						moveEffects,
-						CrossFieldTarget.Source,
-						newCurrMark.revision,
-						newCurrMark.id,
-					);
-					effect.detacher = baseIntention;
-					effect.pairedMarkStatus = PairedMarkUpdate.Reactivated;
-
-					// TODO: Rebase newCurrMark's changes over baseMark's changes.
-					return newCurrMark;
-				} else if (newCurrMark.type === "Revive" && !newCurrMark.isIntention) {
-					assert(
-						isSkipLikeReattach(newCurrMark),
-						0x4ff /* Only a skip-like reattach can overlap with a ReturnFrom */,
-					);
-					// The already populated cells that currMark aimed to revive content into
-					// are having their contents detached by baseMark.
-					// The revive mark remains conflicted because the detach was performed by a different change than
-					// the change the revive aims to revert.
-					// After this, the only way for the reattach to to recover from the conflict is for the nodes to be
-					// returned and for the original deletion (currMark.detachedBy) to be re-applied.
-					// Update the characterization of the deleted content
-					newCurrMark.lastDetachedBy = baseIntention;
-					newCurrMark.detachIndex = baseInputOffset;
-
-					// TODO: Rebase newCurrMark's changes over baseMark's changes.
-					return newCurrMark;
-				}
-			}
+function rebaseNodeChange<TNodeChange>(
+	currMark: Mark<TNodeChange>,
+	baseMark: Mark<TNodeChange>,
+	nodeRebaser: NodeChangeRebaser<TNodeChange>,
+): Mark<TNodeChange> {
+	const baseChange = getNodeChange(baseMark);
+	const currChange = getNodeChange<TNodeChange>(currMark);
 
-			// TODO: Rebase currMark's changes over baseMark's changes.
-			// Note that even if currMark is a skip mark,
-			// it could become a modify mark after rebasing its (empty) changes over baseMark's changes.
-			if (!isSkipMark(currMark)) {
-				getOrAddEffect(
-					moveEffects,
-					CrossFieldTarget.Destination,
-					baseMark.revision ?? baseRevision,
-					baseMark.id,
-				).movedMark = cloneMark(currMark);
-			}
-			return 0;
-		}
-		default:
-			fail(`Unsupported mark type: ${baseType}`);
+	return withNodeChange(currMark, nodeRebaser(currChange, baseChange));
+}
+
+function makeDetachedMark<T>(
+	mark: Skip | ExistingCellMark<T>,
+	detachIntention: RevisionTag,
+	offset: number,
+): Mark<T> {
+	if (isSkipMark(mark)) {
+		return 0;
 	}
+
+	assert(mark.detachEvent === undefined, "Expected mark to be attached");
+	return { ...mark, detachEvent: { revision: detachIntention, index: offset } };
+}
+
+function withoutDetachEvent<T, TMark extends ExistingCellMark<T>>(mark: TMark): TMark {
+	const newMark = { ...mark };
+	delete newMark.detachEvent;
```

Comment:
This function does not clear the lineage. This could lead to a mark that has lineage despite referring to a filled cell.